vol_dataset_test.c

```diff
@@ -3989,11 +3989,210 @@ test_dataset_property_lists(void)
+        H5_FAILED();
+        HDprintf("    couldn't open container group '%s'\n", DATASET_TEST_GROUP_NAME);
+        goto error;
+    }
+
+    if ((group_id = H5Gcreate2(container_group, DATASET_STORAGE_SIZE_TEST_GROUP_NAME, H5P_DEFAULT,
+                               H5P_DEFAULT, H5P_DEFAULT)) < 0) {
+        H5_FAILED();
+        HDprintf("    couldn't create container sub-group '%s'\n", DATASET_STORAGE_SIZE_TEST_GROUP_NAME);
+        goto error;
+    }
+
+    /* TODO: If writes are needed to force allocation, then replace this with a simple type */
+    if ((type_id = DATASET_STORAGE_SIZE_TEST_TYPE) < 0)
+        TEST_ERROR;
+
+    if ((space_id = H5Screate_simple(DATASET_STORAGE_SIZE_TEST_ALL_DSET_SPACE_RANK, dims, NULL)) < 0)
+        TEST_ERROR;
+
+    if ((dcpl_id = H5Pcreate(H5P_DATASET_CREATE)) < 0)
+        TEST_ERROR;
+
+    if ((H5Pset_layout(dcpl_id, H5D_CONTIGUOUS)) < 0)
+        TEST_ERROR;
+
+    if ((dset_id_contiguous = H5Dcreate(group_id, DATASET_STORAGE_SIZE_TEST_DSET_CONTIGUOUS_NAME, type_id,
+                                        space_id, H5P_DEFAULT, dcpl_id, H5P_DEFAULT)) < 0)
+        TEST_ERROR;
+
+    if ((H5Pset_layout(dcpl_id, H5D_CHUNKED)) < 0)
+        TEST_ERROR;
+
+    if ((H5Pset_chunk(dcpl_id, DATASET_STORAGE_SIZE_TEST_ALL_DSET_SPACE_RANK, dims)) < 0)
+        TEST_ERROR;
+
+    if ((dset_id_chunked = H5Dcreate(group_id, DATASET_STORAGE_SIZE_TEST_DSET_CHUNKED_NAME, type_id, space_id,
+                                     H5P_DEFAULT, dcpl_id, H5P_DEFAULT)) < 0)
+        TEST_ERROR;
+
+    if ((H5Pset_deflate(dcpl_id, 6)) < 0)
+        TEST_ERROR;
+
+    if ((dset_id_filtered = H5Dcreate(group_id, DATASET_STORAGE_SIZE_TEST_DSET_FILTERED_NAME, type_id,
+                                      space_id, H5P_DEFAULT, dcpl_id, H5P_DEFAULT)) < 0)
+        TEST_ERROR;
+
+    if ((type_size = H5Tget_size(type_id)) <= 0)
+        TEST_ERROR;
+
+    if ((write_buf =
+             calloc(DATASET_STORAGE_SIZE_TEST_ALL_DSET_EXTENT * DATASET_STORAGE_SIZE_TEST_ALL_DSET_EXTENT,
+                    type_size)) == NULL)
+        TEST_ERROR;
+
+    for (size_t i = 0; i < DATASET_STORAGE_SIZE_TEST_ALL_DSET_EXTENT; i++)
+        for (size_t j = 0; j < DATASET_STORAGE_SIZE_TEST_ALL_DSET_EXTENT; j++)
+            ((int *)write_buf)[i * DATASET_STORAGE_SIZE_TEST_ALL_DSET_EXTENT + j] = j;
+
+    PASSED();
+
+    BEGIN_MULTIPART
+    {
+        PART_BEGIN(H5Dget_storage_size_contiguous)
+        {
+            TESTING_2("H5Dget_storage_size on contiguous dataset");
+
+            if (H5Dwrite(dset_id_contiguous, type_id, H5S_ALL, H5S_ALL, H5P_DEFAULT,
+                         (const void *)write_buf) < 0) {
+                H5_FAILED();
+                HDprintf("    couldn't write to contiguous dataset\n");
+                PART_ERROR(H5Dget_storage_size_contiguous);
+            }
+
+            if (H5Dget_storage_size(dset_id_contiguous) <= 0) {
```

Comment:
Should only check == 0 for these calls. Since the return value is an hsize_t, this will probably generate compiler warnings about comparing an unsigned to a negative value